### DISCUSSION TITLE
Can no longer load migrations after setup in testbench-core 8.23 (testbench 8.22)

### DISCUSSION THREAD

**Discussion by Rockylars** [orchestral/testbench-core#194] [General]
I used a very simple testclass extension where i did the following, though i simplified it here.
```php
use Orchestra\Testbench\TestCase;
abstract class AsApplication extends TestCase
{
    use RefreshDatabase;
    protected function setUp(): void
    {
        parent::setUp();
        $this->loadMigrationsFrom(__DIR__ . '/../../../Fake/Migrations');
    }
}
```
This way i can load migrations for specific tests and when i need external tables for my package tests.

But since 8.23 with `load_migration_paths()` added it no longer works, unless i use the following approach.
```php
use Orchestra\Testbench\TestCase;
abstract class AsApplication extends TestCase
{
    use RefreshDatabase;
    protected function setUp(): void
    {
        parent::setUp();
    }

    protected function defineDatabaseMigrations(): void
    {
        $this->loadMigrationsFrom(__DIR__ . '/../../../Fake/Migrations');
    }
}
```
But i would still like to just load them in how they worked originally, without having to `parent::` chain the defineDatabaseMigrations call, as i now can't easily do it per test.

**Comment by crynobone** [comment]
You should update to use the correct method for future compatibility with `RefreshDatabase` and in-memory database for L11.

**Reply by Rockylars** [reply]
But this is a BC break that removes the ability to load migrations per test, I'll try to upgrade once I figure out why my GitHub Actions seems to load them before while my local loads them after the package's migrations, something something Composer problems? Not sure.

**Reply by crynobone** [reply]
I was never intended that to be used on `seTup()` nor want to cover that under any BC.

https://packages.tools/testbench/the-basic/databases.html

**Reply by Rockylars** [reply]
That is true, but it is sad news. Loading migrations at any point or simply right after the package migrations makes test setups very flexible. As far as i can tell, these migrations are loaded before the ones of the package, is there a way to schedule test (class) specific migrations after the package migrations?

**Reply by crynobone** [reply]
Using `RefreshDatabase` with running migrations per test only worked in Testbench 8 and below previously due to the way that `:in-memory:` database will reset the database between tests. This is no longer the case for Laravel 11 and the new changes ensure you can upgrade your packages to support Laravel 11 while tests still work consistently between multiple Laravel versions.

**Reply by crynobone** [reply]
This also makes `RefreshDatabase` in Laravel 11 work the same when using `in-memory` and persistent databases.

**Reply by Rockylars** [reply]
Big shame, It does make it run faster but makes it a lot less flexible, I'll try to see if i can find a way to run tests with different migrations loaded in somehow, otherwise I'll have to just run them all with the isolated option.

**Reply by crynobone** [reply]
Test should work fine even without `RefreshDatabase` or you should consider using `DatabaseMigrations` instead. There's also some method available after database has been refreshed (see `RefreshDatabase` trait) but again doesn't work well if migrations does change between tests (in single TestCase class).

**Reply by Rockylars** [reply]
That's true, I even remember looking at those this morning and just immediately forgot about them. It's no big deal, I'll play around with it until I find a way to gain my preferred way of testing back, thanks for the help and kindness so far though! :smiley:

**Comment by Rockylars** [comment]
It seems to be working again when i directly upgraded to version 9 + Laravel 11, thank you at least! It seems v8.23.3 probably was the one that fixed it.
